src/Umbraco.Cms.Integrations.Search.Algolia/App_Plugins/UmbracoCms.Integrations/Search/Algolia/js/algolia.service.js

```diff
@@ -1,17 +1,32 @@
 ﻿function algoliaService(contentTypeResource) {
     return {
         getContentTypes: function (callback) {
+
             contentTypeResource.getAll().then(function (data) {
-                callback(data.filter(item => item.parentId == -1 && !item.isElement).map((item) => {
-                    return {
-                        id: item.id,
-                        icon: item.icon,
-                        alias: item.alias,
-                        name: item.name,
-                        selected: false,
-                        allowRemove: false
+
+                var contentTypesArr = [];
+
+                for (let i = 0; i < data.length; i++) {
+                    
+                    if (data[i].isElement) continue;
+
+                    contentTypeResource.getWhereCompositionIsUsedInContentTypes(data[i].id).then(function (response) {
+                        if (response.length === 0) {
+                            contentTypesArr.push({
+                                id: data[i].id,
+                                icon: data[i].icon,
+                                alias: data[i].alias,
+                                name: data[i].name,
+                                selected: false,
+                                allowRemove: false
+                            });
+                        }
+                    });
+
+                    if (data.length - 1 === i) {
```

Comment:
Could you not just call `callback(contentTypesArr)` from outside the loop, instead of checking if you are on the last element from within the loop?